### DISCUSSION TITLE
Filter out results based on Email Address/Username

### DISCUSSION THREAD

**Discussion by Id10tSupport** [YannickRe/spfx-msgraph-peoplesearch#56] [Q&A] [answered]
What is the recommended method to filter out results based on specific characters in the email address or UPN?

In particular, we use _ and a 3 letter designator at the end of accounts (in Active Directory) to identify them as admin accounts, privileged access accounts, or service accounts.  username_xxx@domain.com

I would like to be able to return user results that do not contain these types of accounts.

**Comment by YannickRe** [comment]
The format you use makes it hard to filter them out, as there is no support for `contains` or `endsWith` in `$filter` for the Users endpoint. If it'd be xxx_username@domain.com, you could have used `startsWith`.
In this case, unless you have another property you could filter on, I'm afraid I cannot help you.

More info on the options, can be found at:

-  https://docs.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-1.0&WT.mc_id=M365-MVP-5003400
- https://docs.microsoft.com/en-us/graph/query-parameters?view=graph-rest-1.0&WT.mc_id=M365-MVP-5003400

**Comment by Id10tSupport** [comment]
What properties are available for filtering?

**Reply by YannickRe** [reply]
You can check that on the first link, every property is there and is marked with "Supports `$filter`" when it supports filtering => https://docs.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-1.0&WT.mc_id=M365-MVP-5003400#properties

**Comment by Id10tSupport** [comment, marked as answer]
I found that the following works:
not endswith(userPrincipalName,'_xxx@lazard.com') and not endswith(userPrincipalName,'_xxx@lazard.com')

**Reply by YannickRe** [reply]
Good to hear you found a solution!
